lib/components/base-components/NormalComponent.ts

```diff
@@ -98,20 +98,23 @@ export class NormalComponent<
     // Handle schPortArrangement
     const schPortArrangement = this._parsedProps.schPortArrangement as any
     if (schPortArrangement) {
+      let pinNumber = 1
       for (const side in schPortArrangement) {
         const pins = schPortArrangement[side].pins
         if (Array.isArray(pins)) {
-          for (const pinNumber of pins) {
-            portsToCreate.push(
-              new Port(
-                {
-                  pinNumber,
-                },
-                {
-                  originDescription: `schPortArrangement:${side}`,
-                },
-              ),
-            )
+          for (const pin of pins) {
+            // check if pinNumber is a number or a string
+            const pinNumberInt = parseInt(pin)
+            if(pinNumberInt) {
+              portsToCreate.push(
+                new Port({ pinNumber: pinNumberInt }),
+              )
+            } else {
+              portsToCreate.push(
+                new Port({ pinNumber: pinNumber, aliases: [pin] }),
+              )
+            }
+            pinNumber++
```

Comment:
This logic isn't sound in many ways

- doesn't handle `"pin1"`
- doesn't handle many number parsing cases properly (use `/\d+/.match` to determine if a string is a number). As is, stuff like 1.1 will cause issues.
- depending on the order in which things are defined, duplicate numbers are added

Reply:
@imrishabh18 copy this function into core and export it, it's a useful function that handles virtually every case properly. We can pre-process `pins` with it.

When you copy over this function can you also copy it's tests, we can then patch easyeda-converter to import it when the function gets merged into core

https://github.com/tscircuit/easyeda-converter/blob/main/lib/normalize-pin-labels.ts